# **Análise e predição do posicionamento político do eleitor brasileiro**

Projeto final realizado para a disciplina de Ciência de Dados do curso de Ciência da Computação, *Universidade de São Paulo*.

## **Introdução**

O presente trabalho tem como objetivo realizar, primeiramente, uma *análise exploratória* dos dados provenientes de uma pesquisa de opinião sobre o perfil ideológico dos eleitores brasileiros, feita pelo Instituto DataSenado no ano de 2024. O conjunto de dados original contém diversas variáveis de natureza sociodemográfica, comportamental e política, mas nem todas serão utilizadas. Serão mantidas somente aquelas que possam contribuir para compreender a variável alvo (*target*) **Posicionamento Político**, que representa o espectro político autodeclarado por cada participante. A análise de dados buscará, portanto, investigar as relações entre a variável target e as demais variáveis selecionadas, por meio de visualizações gráficas e métodos estatísticos. 

Após a fase exploratória, o projeto avançará para a segunda etapa: a construção de um modelo preditivo. Com o uso de técnicas de *machine learning*, o objetivo será desenvolver um modelo capaz de prever o espectro político com base nas respostas às variáveis selecionadas da pesquisa.

Importante mencionar que este trabalho não se preocupará em fazer inferência sobre a preferência política da população brasileira como um todo, tendo em vista que uma parcela significativa dos dados originais não será utilizada.

* <small>Os dados utilizados neste trabalho foram obtidos por meio dos microdados disponibilizados pelo Instituto DataSenado em <a href="https://www12.senado.leg.br/institucional/datasenado/publicacaodatasenado?id=pesquisa-traca-perfil-ideologico-dos-eleitores-brasileiros" target="_blank" rel="noopener noreferrer">Pesquisa traça perfil ideológico dos eleitores brasileiros</a>.</small>
* <small>Junto com os dados foi disponibilizado um dicionário para auxiliar na interpretação.</small>

## **1. Importação de bibliotecas**

In [8]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## **2. Carregamento dos dados**

In [9]:
df = pd.read_csv(r'data\raw\panorama_politico_06_2024\DATASEN BR 2024 06 BAROMETRO - DADOS.csv', sep=';')

## **3. Visão geral dos dados não tratados**

In [10]:
df.shape

(21808, 109)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21808 entries, 0 to 21807
Columns: 109 entries, ID to W2
dtypes: float64(49), int64(52), object(8)
memory usage: 18.1+ MB


In [12]:
df.head()

ID  P04  P06  P07  P08  P09  P10  P11  P12  P13  ...  VD_EDUCACAO  \
0   1   41    3    3    4    2    2    2    2    1  ...            4   
1   2   29    2    2    2    2    2    2    2    2  ...            3   
2   3   28    1    2    2    2    2    2    2    2  ...            3   
3   4   28    3    2    2    2    1    2    2    2  ...            1   
4   5   24    1    2    1    3    2    2    2    1  ...            3   

        POP_UF_SEXO       POP_UF_RACA      POP_UF_IDADE  POP_UF_RESIDENCIA  \
0  4599548,01319187  5928889,83043946  1736626,25524565    8334552,9943779   
1  6203186,58563845  9446410,89141747  2251115,21205504   8765141,45702551   
2   966058,41748564  1406439,72866629   527384,65232656    371734,75555842   
3   966058,41748564  1406439,72866629   374974,77422564   1462103,48138951   
4  1345034,43392631  1067871,54843917   582338,13699172   2344797,51342096   

   POP_UF_PORTE   POP_UF_TRABALHO        POP_UF_EDU                 W1  \
0       4791536  5981782,45942209  2337434,74097726   2,79749300778953   
1       3033950   970710,50572401  3842125,61215856   1,12345379164713   
2        998303   995087,55192826   527747,57235541  0,335079450803203   
3        608944   728396,78367523   679620,91966034  0,335079450803203   
4        973772  1378587,88363752   865902,52059461  0,242935105861798   

                 W2  
0  12549,6687231227  
1  2805,47489557209  
2  3533,35524432939  
3  16466,2467766673  
4  2386,70789580883  

[5 rows x 109 columns]

O conjunto de dados inicialmente contém 21.808 linhas e 109 colunas (perguntas). Cada linha é a resposta de um entrevistado para as 109 perguntas. É possível observar que tanto as perguntas quanto as respostas estão em formato numérico. Dessa forma, para facilitar o entendimento e a análise dos dados, cabe converter esses valores para a forma textual.